media/js/base/stub-attribution.js

```diff
@@ -0,0 +1,264 @@
+
+    /**
+     * Returns stub attribution value used for rate limiting.
+     * @return {Number} float between 0 and 1.
+     */
+    StubAttribution.getAttributionRate = function() {
+        var rate = $('html').attr('data-stub-attribution-rate');
+        return isNaN(rate) ? 0 : Math.min(Math.max(parseFloat(rate), 0), 1);
+    };
+
+    /**
+     * Returns true if both cookies exist.
+     * @return {Boolean} data.
+     */
+    StubAttribution.hasCookie = function() {
+        return Mozilla.Cookies.hasItem(StubAttribution.COOKIE_CODE_ID) && Mozilla.Cookies.hasItem(StubAttribution.COOKIE_SIGNATURE_ID);
+    };
+
+    /**
+     * Stores a cookie with stub attribution data values.
+     * @param {Object} data - attribution_code, attribution_sig.
+     */
+    StubAttribution.setCookie = function(data) {
+
+        if (!data.attribution_code || !data.attribution_sig) {
+            return;
+        }
+
+        // set cookie to expire in 24 hours
+        var date = new Date();
+        date.setTime(date.getTime() + (1 * 24 * 60 * 60 * 1000));
+        var expires = date.toUTCString();
+
+        Mozilla.Cookies.setItem(StubAttribution.COOKIE_CODE_ID, data.attribution_code, expires, '/');
+        Mozilla.Cookies.setItem(StubAttribution.COOKIE_SIGNATURE_ID, data.attribution_sig, expires, '/');
+    };
+
+    /**
+     * Gets stub attribution data from cookie.
+     * @return {Object} - attribution_code, attribution_sig.
+     */
+    StubAttribution.getCookie = function() {
+        return {
+            /* eslint-disable camelcase */
+            attribution_code: Mozilla.Cookies.getItem(StubAttribution.COOKIE_CODE_ID),
+            attribution_sig: Mozilla.Cookies.getItem(StubAttribution.COOKIE_SIGNATURE_ID)
+            /* eslint-enable camelcase */
+        };
+    };
+
+    /**
+     * Updates all download links on the page with additional query params for
+     * stub attribution.
+     * @param {Object} data - attribution_code, attribution_sig.
+     */
+    StubAttribution.updateBouncerLinks = function(data) {
+        /**
+         * If data is missing or the browser does not meet requirements for
+         * stub attribution, then do nothing.
+         */
+        if (!data.attribution_code || !data.attribution_sig || !StubAttribution.meetsRequirements()) {
+            return;
+        }
+
+        // target download buttons and other-platforms modal links.
+        $('.download-list .download-link, .download-platform-list .download-link').each(function() {
+            var version;
+            // If this is a transitional download link do nothing.
+            if (this.href && this.href.indexOf('/firefox/new/?scene=2') === -1) {
+
+                version = $(this).data('downloadVersion');
+                // Currently only Windows 32bit uses the stub installer, but this could
+                // easily change in the future so we'll make that bet now.
+                if (version && (version === 'win' || version === 'win64')) {
```

Comment:
Hmm, no - but perhaps my comment didn't make sense. I'm betting that they will want attribution data for both 32 and 64 bit builds eventually (even though it only forks with 32 bit installer currently), so I'm adding that case now. I'll update the comment to make it a little clearer.